### ISSUE TITLE
Gradle 6.4 and JPMS, error: module kotlin.stdlib cannot be found in the module graph

### ISSUE THREAD

**Issue by elect86** [Kotlin/dokka#851]
**Describe the bug**
Trying to implement Gradle 6.4, JPMS, Kotlin and Dokka, whenever I run `build` I get in the console:
> error: module kotlin.stdlib cannot be found in the module graph

docs seems fine though

**To Reproduce**
[Example project](https://github.com/kotlin-graphics/kotlin-unsigned/tree/jpms)

**Dokka configuration**

```kotlin
tasks {
    val dokka by getting(org.jetbrains.dokka.gradle.DokkaTask::class) {
        outputFormat = "html"
        outputDirectory = "$buildDir/dokka"
    }
}

val dokkaJar by tasks.creating(Jar::class) {
    group = JavaBasePlugin.DOCUMENTATION_GROUP
    description = "Assembles Kotlin docs with Dokka"
    archiveClassifier.set("javadoc")
    from(tasks.dokka)
}

artifacts {
    archives(dokkaJar)
}
```

**Installation**
- Operating system: Win10 x64
- Build tool: Gradle v6.4
- Dokka version: 0.10.1

Also, there some stuff which are gonna be deprecated in Gradle 7.0

if I run `build --warning-mode all'...`

> Task :dokka
Property 'dokkaRuntime' is not annotated with an input or output annotation. This behaviour has been deprecated and is scheduled to be removed in Gradle 7.0. See https://docs.gradle.org/6.4-rc-1/userguide/more_about_tasks.html#sec:up_to_date_checks for more details.
Property 'configuration.collectKotlinTasks' is not annotated with an input or output annotation. This behaviour has been deprecated and is scheduled to be removed in Gradle 7.0. See https://docs.gradle.org/6.4-rc-1/userguide/more_about_tasks.html#sec:up_to_date_checks for more details.
Property 'configuration.name' is not annotated with an input or output annotation. This behaviour has been deprecated and is scheduled to be removed in Gradle 7.0. See https://docs.gradle.org/6.4-rc-1/userguide/more_about_tasks.html#sec:up_to_date_checks for more details.
logging: loading modules: [com.github.kotlin_graphics.kotlin_unsigned, java.base, kotlin.stdlib.jdk8]
error: module kotlin.stdlib.jdk8 cannot be found in the module graph
C:\Users\elect\IdeaProjects\kotlin-unsigned\src\main\java\module-info.java: error: the Kotlin standard library is not found in the module graph. Please ensure you have the 'requires kotlin.stdlib' clause in your module definition

**Comment by elect86** [comment]
Hi,

any news on this?

**Comment by lion7** [comment]
Just had the same issue. 
In my case I had the `module-info.java` file in `src/main/java`  and my Kotlin sources in `src/main/kotlin`.
Moving the `module-info.java` file to `src/main.kotlin` solved the error.

**Comment by elect86** [comment]
> Moving the `module-info.java` file to `src/main/kotlin` solved the error.

Weird, it shouldnt make any sense that..

**Comment by kad-leeuwg1** [comment]
Nope, to be honest it felt kinda odd that it solved the problem.
I was thinking that maybe the sources in `src/main/java` are not picked up by the Kotlin compiler.
Or that the Java compile attempts to compile only the `module-info.java` and is ignoring the Kotlin sources.
No clue...

**Comment by elect86** [comment]
Which version are you using? I solved with 1.4.0
